pkg/controller/manager/manager_controller.go

```diff
@@ -327,15 +328,16 @@ func (r *ReconcileManager) Reconcile(ctx context.Context, request reconcile.Requ
 		return reconcile.Result{}, err
 	}
 
-	trustedSecretNames := []string{render.PacketCaptureCertSecret, monitor.PrometheusTLSSecretName, relasticsearch.PublicCertSecret, render.ProjectCalicoApiServerTLSSecretName(installation.Variant)}
-
+	// Determine if compliance is enabled.
 	complianceLicenseFeatureActive := utils.IsFeatureActive(license, common.ComplianceFeature)
 	complianceCR, err := compliance.GetCompliance(ctx, r.client)
 	if err != nil && !errors.IsNotFound(err) {
 		r.status.SetDegraded(operatorv1.ResourceReadError, "Error querying compliance: ", err, reqLogger)
 		return reconcile.Result{}, err
 	}
 
+	// Build a trusted bundle containing all of the certificates of components that communicate with the manager pod.
+	trustedSecretNames := []string{render.PacketCaptureCertSecret, monitor.PrometheusTLSSecretName, relasticsearch.PublicCertSecret, render.ProjectCalicoApiServerTLSSecretName(installation.Variant)}
```

Comment:
I am not sure it needs it, because CreateTrustedBundle already includes the Calico root certificate which can be used to verify any certs signed by this operator. 

I'm not 100% sure how all of this logic works though, so maybe we should have a chat offline?

Reply:
You are right, but you are creating a gap in functionality though. The certificate management stuff is designed in a way that each tls secret can be replaced by a user with their own by dropping it in the operator namespace. When a controller runs it reads the key pair from the operator namespace and copies it over to the target namespace. You are missing this part. Also the certificate management is taking care of the installation.spec.certificateManagement feature.